flow/scripts/report_metrics.tcl

```diff
@@ -116,12 +116,12 @@ proc report_metrics { when {include_erc true} {include_clock_skew true} } {
     puts "\n=========================================================================="
     puts "$when setup_violation_count"
     puts "--------------------------------------------------------------------------"
-    puts "setup violation count [llength [find_timing_paths -path_delay max -slack_max 0]]"
+    puts "setup violation count [llength [find_timing_paths -path_delay max -slack_max 0 -group_count $sta::group_count_max]]"
```

Comment:
@jjcherry56 is this the best way to do this?  It seems like we want the number of violating endpoints rather than the number of paths but I'm not sure how to get that.

Reply:
@tspyrou do you know how best to do this?

Reply:
@maliberty iterate the endpoint pins and count the ones with slack < 0. I think this is exposed to tcl.

Reply:
https://github.com/The-OpenROAD-Project/OpenSTA/blob/333b91e72d8b87fe442f0608cc5c73c741e18dda/tcl/StaTcl.i#L277

Reply:
@antonblanchard can you make this change?

Reply:
Thanks @tspyrou. This is what I've come up with:

```
proc count_violating_endpoints { minmax } {
    set count 0
    foreach pin [sta::endpoints] {
        foreach vertex [$pin vertices] {
            set slack [ $vertex slack $minmax ]
            if [ expr $slack < 0 ] {
                set count [expr $count + 1]
            }
        }
    }
    return $count;
}
```

Reply:
That's right, although you can use incf instead of expr to bump the count.
Coming to an STA near you is the function sta::endpoint_violation_count which does this is c++ instead of tcl.
Just have to get through the CI hurdles.

Reply:
Thanks @jjcherry56!